docs/cudf/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,288 @@
+then have that benchmark automatically run for all objects of interest.
+
+### Custom fixtures
+
+Developers may define custom fixtures if necessary, but this should be done with care.
+The `benchmark_with_object` decorator covers most use cases and automatically improves benchmark coverage.
+When writing fixtures, developers should make the data sizes dependent on the benchmarks configuration.
+The `benchmarks/common/config.py` file defines standard data sizes to be used in benchmarks.
+These data sizes can be tweaked for debugging purposes (see {ref}`testing` below).
+Fixture sizes should be relative to the `NUM_ROWS` and/or `NUM_COLS` variables defined in the config module.
+
+### Parametrization vs. fixtures
+
+Our benchmarks make use of the [`pytest_cases`](https://smarie.github.io/python-pytest-cases/) `pytest` plugin.
+This plugin allows us to handle parametrization much more cleanly than pytest does out of the box.
+Specifically, it provides some syntactic sugar around
+
+```python
+@pytest.mark.parametrize(
+    "num", [1, 2, 3]
+)
+def bench_foo(benchmark, num):
+    benchmark(num * 2)
+```
+
+for when the parameters are nontrivial and require complex initialization.
+This is common for benchmarks of functions accepting cuDF objects, such as `cudf.concat`.
+With `pytest_cases`, the different cases are instead placed into separate functions and automatically made available.
+
+```python
+# bench_foo_cases.py
+def case_1():
+    return 1
+
+def case_2():
+    return 2
+
+def case_3():
+    return 3
+
+# bench_foo.py
+@pytest_cases.parametrize_with_cases(num)
+def bench_foo(benchmark, num):
+    benchmark(num * 2)
+```
+
+This approach is strongly encouraged within cuDF benchmarks.
+It forces developers to put complex initialization into named, documented functions.
+That becomes especially valuable when benchmarking APIs whose performance can vary drastically based on parameters.
+Additionally, cases, like fixtures, are lazily evaluated.
+Initializing complex objects inside a `pytest.mark.parametrize` can dramatically slow down test collection,
+or even lead to out of memory issues if too many complex cases are collected.
+Using lazy case functions ensures that the associated objects are only created on an as-needed basis.
+
+When writing cases, just as in writing custom fixtures, developers should be make use of the config variables.
+Cases should import the `NUM_ROWS` and/or `NUM_COLS` variables from the config module and use them to define data sizes.
+
+The observant reader may recognize that cases seem quite familiar to fixtures.
+While the implementations are in fact quite similar, for our purposes it is important to keep the two distinct.
+From a conceptual standpoint, fixtures are homogeneous and generic while cases are heterogeneous and specific.
+Fixtures should be created sparingly and used broadly.
+Cases can be precisely tailored for a single test.
```

Comment:
I think I don't understand this paragraph. At least, it doesn't give me a good idea when I should be using cases over fixtures (or vice versa).

Reply:
This discussion is being continued in much greater detail [here](https://github.com/rapidsai/cudf/pull/11180#discussion_r918780717).

Reply:
I've adapted the conclusions of that discussion into a clear set of guidelines in the latest version of this document.